src/mail.sh

```diff
@@ -115,8 +116,14 @@ function mail_send()
     cmd+=" --cc=\"$cc_recipients\""
   fi
 
-  pre_generate_patches "$commit_range" "$version"
-  patch_count="$(count_patches "$patch_cache")"
+  if [[ -z "$patch_source" ]]; then
+    pre_generate_patches "$commit_range" "$version"
+    patch_count="$(count_patches "$patch_cache")"
+  elif [[ -d "$patch_source" ]]; then
+    patch_count="$(count_patches "$patch_source")"
+  elif [[ $(is_a_patch "$patch_source") ]]; then
+    patch_count=1
+  fi
```

Comment:
Add a comment inside each if that highlight the case the if condition is handling.